### DISCUSSION TITLE
How to build a infiniband PXE via ipxe

### DISCUSSION THREAD

**Discussion by crayflames** [ipxe/ipxe#1272] [Q&A]
Hi all

There are a question that I try to build a infiniband PXE but no any idea and  i can't find resource google.

I had install ipxe on my server and Client boot to ipxe with ethernet mode is successed.

But I change my CX7 (We use port1 of this card as DHCP port )to IB mode and restart dhcp is failed.

Are there something setting I need to modify ??

Thanks for your answer

**Comment by NiKiZe** [comment]
ipxe.efi and ipxe.kpxe should be full feature, which binary/build/command did you use to build your current iPXE binary?

**Reply by crayflames** [reply]
Hi NiKiZe

I had already compiler ipxe.efi by following https://ipxe.org/download.

If I change the dhcp config file "/etc/default/dhcp" => "IPv4=ib0"

Then restsrt ise-dhcp-server by systemctl restart isc-dhcp-server . The step will be failed

Did I misunderstand something?

**Reply by NiKiZe** [reply]
If you have issues configuring your DHCP server I can't really help, I wouldn't expect any DHCP setting to change which wire protocol a NIC uses.

DHCP will control layer 3, not layer 1-2

**Reply by crayflames** [reply]
Thanks for clarified something I dont understand
